### PR TITLE
ICU-22194 Add CI job to generate Github Pages using Github Actions

### PR DESCRIPTION
**Pull request by echeran**
## Background

Our User Guide is published via the Github Pages feature. I noticed that updates to it have been [failing over the past 2 weeks](https://github.com/unicode-org/icu/actions/workflows/pages/pages-build-deployment).

When we set it up a couple of years ago, it was still a black box how Github would run its internal version of Jekyll to generate the static pages. Since then, the Github Actions CI framework has become popular, and it is now the [default place to generate and/or deploy Github Pages](https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/). Even the legacy mechanism is now represented as a Github Actions CI workflow, as the links above show and describe.

## Changes in this PR

This PR introduces a new CI workflow that supports the deployment of our Github Pages (User Guide).

If the new CI workflow succeeds as expected, then we should follow [these instructions](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow) to update the repo's Github settings for how to publish our Github Pages. Making that update will use the new CI workflow in favor of the currently broken legacy black box style of generating Github Pages.

Note: Figuring out the new workflow was very tricky because, on top of everything else, the Jekyll theme that we use will occasionally have conflicts with the versions of Jekyll-related dependencies that Github likes to use. It appears that the latest 0.4.1 and 0.4.2 versions of our Jekyll theme, which were released in early February, have a parser error that manifests when combined into our GH Pages Jekyll setup, and that is what caused our Github Pages to stop updating. Pinning the version of the theme to 0.4.0 gets it to work again.

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22194
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by markusicu — DISMISSED** [review]
Thank you so much for figuring this out!!

File: .github/workflows/jekyll-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# build instead of relying on the default behavior from
+# https://github.com/actions/jekyll-build-pages . The default behavior of 
+# actions/jekyll-build-pages seems to be causing errors with this theme.
+
+name: Deploy User Guide
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (echeran) [reply .github/workflows/jekyll-gh-pages.yml]: Yes, it turns out. I just applied that change to this PR based on the [info I found](https://stackoverflow.com/a/63461325/2077918).

**Comment by echeran** [comment]
I also [found out from the maintainers of the Jekyll theme](https://github.com/just-the-docs/just-the-docs/discussions/1196) that we use that we had a stray file from the theme in our docs directory that was causing the problems. I applied the change here as well.

**Review by markusicu — APPROVED** [review]
lgtm tnx!

**Pull request merged**